lighthouse-core/scripts/compare-runs.js

```diff
@@ -247,17 +255,20 @@ function aggregateResults(name) {
 }
 
 /**
- * @param {*[]} results
+ * @param {Array<{name: string}>} results
  */
 function filter(results) {
   const includeFilter = argv.filter ? new RegExp(argv.filter, 'i') : null;
 
   results.forEach((result, i) => {
     for (const propName of Object.keys(result)) {
-      if (reportExcludeRegex && reportExcludeRegex.test(propName)) delete result[propName];
+      if (reportExcludeRegex && reportExcludeRegex.test(propName)) {
+        // @ts-ignore: propName is a key.
+        delete result[propName];
+      }
     }
 
-    if (includeFilter && !includeFilter.test(result.key)) {
+    if (includeFilter && !includeFilter.test(result.name)) {
```

Comment:
keeping it on key meant you can do `--filter=metric` (which is documented and i really like)

we could use something else for metric/timing filtering vs this regex on name tho.

Reply:
ah, just had to move this filter check before the property delete :)